CMakeLists.txt

```diff
@@ -20,14 +20,24 @@ option(USE_VCPKG "Use VCPKG to download and manage dependencies" OFF)
 if (USE_VCPKG)
     # Detect correct triplet based on CMake env
     if (OSXCROSS_TARGET)
-        if(CMAKE_OSX_ARCHITECTURES STREQUAL "x86_64")
-            set(VCPKG_TARGET_TRIPLET "x64-osxcross")
-        elseif(CMAKE_OSX_ARCHITECTURES STREQUAL "arm64")
-            set(VCPKG_TARGET_TRIPLET "arm64-osxcross")
+        if (IOS)
+            if(CMAKE_OSX_ARCHITECTURES STREQUAL "arm64")
+                set(VCPKG_TARGET_TRIPLET "arm64-ioscross")
+            endif()
+            set(VCPKG_CHAINLOAD_TOOLCHAIN_FILE ${CMAKE_CURRENT_SOURCE_DIR}/platform/ioscross/ios.cmake)
+            set(VCPKG_CMAKE_SYSTEM_NAME iOS)
+            set(VCPKG_TARGET_ARCHITECTURE "${CMAKE_OSX_ARCHITECTURES}")
+            set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS}  -DIOS=1")
```

Comment:
I think at this point you should create a vcpkg triplet file to encapsulate all these flags and whatever logic is needed

Reply:
I moved the logic to the vcpkg triplet.